### PR TITLE
Add date range filter to platform admin

### PR DESCRIPTION
**Pull request by servingUpAces**
A date range has been added to the platform admin page. 

Be aware that when using this query it looks at the notification_history table, this may results in a slow running query. I will be testing the performance once I get it deployed to production with a large data set. 

screen shot 2017-01-03 at 11 39 07

### CODE REVIEW

File: app/notify_client/service_api_client.py [hunk 1]
```diff
@@ -60,6 +60,7 @@ def get_services(self, params_dict=None):
         """
         Retrieve a list of services.
         """
+        print(params_dict)
```
Comment (leohemsted) [inline app/notify_client/service_api_client.py]: :printer:

File: tests/app/main/views/test_platform_admin.py [hunk 2]
```diff
@@ -103,25 +99,36 @@ def test_platform_admin_toggle_including_from_test_key(
 ):
     with app_.test_request_context():
         with app_.test_client() as client:
+            print(include_from_test_key)
```
Comment (leohemsted) [inline tests/app/main/views/test_platform_admin.py]: :printer:

**Comment by imdadahad** [comment]
@leohemsted I personally try to avoid using date pickers. I've found in my experience that existing date picker libraries aren't that great (not easy to customise and not mobile friendly). I have seen developments by web browsers trying to push towards making them inbuilt but not sure how far that's come with browsers other than Chrome. 

In any case probably better to split the date out into separate fields, easier on the eye and easier for validation too. Checkout https://www.gov.uk/service-manual/design/dates

**Comment by servingUpAces** [comment]
We definitely do not want to use date pickers. 
I don't like the look of these filters, but we are planning to do some more designing around this page and the other MI reports that we are going to need. 
As a first attempt at adding more filters to this page, I think it is ok to leave this as is. Remember this page is only seen by our team. 
I'd like to see how this page actually performs on production where we have more data to deal with.

**Comment by leohemsted** [comment]
@servingUpAces running the load tests is super straightforward - i'd like us to test this on staging first to make sure it doesn't screw up if the system's under load when you request a large page

**Pull request merged**